Switch to using mb_(encode|decode)_mimeheader functions

Pudge601 opened the pull request:
The new "Fixed" section in the change log shows the issues that this change will fix.

Commit:
Add dependency on mbstring extension in composer.json

Commit:
Switch to using `mb_decode_mimeheader` for decoding header values

This should solve issues with dealing with malformed encoded words in headers, as the `mb_decode_mimeheader` will always return some kind of string value that it can determine from the header value.

That behaviour should keep us conforming with RFC 2047 section 6.3, and will allow us to get a working mail object despite borked headers.

Commit:
Switch to using `mb_encode_mimeheader` to encode our headers

This method properly handles creating encoded words, ensuring that we don’t split multi-byte characters between encoded words (RFC 2047 section 3), and ensuring that headers (including the header name) do not exceed 75 characters per line.

Involves a significant (and breaking) change to modify the public `encodeHeaderValue` method to be `encodeHeader`, so that the encode method is able to take into account the header name length when keeping the header values to 75 characters per line.

Commit:
Update unit tests to match the new expected behaviour of encoding headers

We now rightfully expect our headers to be no longer than 75 characters per line.

Commit:
Add notes to change log regarding switching to `mb_` functions for encoding/decoding headers

chrisminett commented on composer.json:
```diff
@@ -25,7 +25,8 @@
     "require": {
         "php": ">=5.6",
         "ext-iconv" : "*",
```
Can we drop requirement for *iconv* now?

Pudge601 replied:
Looks like it, a quick global search for 'iconv' shows up nothing, which I believe means we can get rid of the requirement.

I'll remove this now.

Pudge601 replied:
I've made this change.

Do you think this needs an entry in the change log?

chrisminett commented on tests/__files/attachments-expected-headers.txt:
```diff
@@ -3,20 +3,51 @@ From: From Name <from@example.com>
+ =?UTF-8?B?Qmdrblk4RTZieE5JdElpRGFKOXZIR0x2eU1aaiAgICAgICAgIDZTR2c9PQ==?=
 X-Goomoji-Body: true
 Date: Thu, 16 Aug 2012 15:45:32 +0100
-Message-Id: <CAJ_TRnL=YMObf2FT9bU7NO0ziPYxpnxfxrtOz4r2-aBxkHSOrA@mail.gmail.com>
```
This is a shame. Message-Id would be nice if it were readable.

Pudge601 replied:
Yeah, I noticed this while I was updating these expected outputs..

It is a shame, but not sure what we can do about it to be honest :/

Commit:
Remove requirement for iconv extension in composer.json

We’re no longer using this extension anywhere, so it can be safely removed as a requirement.
